diff --git a/mass_contact.admin.inc b/mass_contact.admin.inc
index 507fe73..7715bde 100644
--- a/mass_contact.admin.inc
+++ b/mass_contact.admin.inc
@@ -70,6 +70,7 @@ function mass_contact_admin_edit(array $form, array $form_state, $cid = NULL) {
   // Initialize the array.
   $edit = array(
     'category' => '',
+    'user_status' => '',
     'recipients' => '',
     'selected' => '',
     'cid' => '',
@@ -93,6 +94,18 @@ function mass_contact_admin_edit(array $form, array $form_state, $cid = NULL) {
     '#required' => TRUE,
   );
 
+  // Create a set of checkboxes, including each role.
+  $form['user_status'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('User status'),
+    '#options' => array(
+      'active' => t('Active'),
+      'blocked' => t('Blocked'),
+    ),
+    '#default_value' => !empty($edit['user_status']) ? unserialize($edit['user_status']) : [1],
+    '#description' => t('Users with those statuses will be included to the mailing list.'),
+  );
+
   $form['recipients'] = array('#tree' => TRUE);
 
   // Add form elements provided by grouping_method plugins.
@@ -103,7 +116,8 @@ function mass_contact_admin_edit(array $form, array $form_state, $cid = NULL) {
     // Get the admin edit function name for this particular implementation.
     $function = ctools_plugin_get_function($plugin, 'mass_contact_admin_edit');
     // Call the plugin function to add the form snippet(s).
-    $form['recipients'][$plugin_name] = $function(unserialize($edit['recipients']));
+    $recipients = !empty($edit['recipients']) ? unserialize($edit['recipients']) : [];
+    $form['recipients'][$plugin_name] = $function($recipients);
   }
 
   $form['selected_categories'] = array(
@@ -207,6 +221,7 @@ function mass_contact_admin_edit_submit(array $form, array &$form_state) {
 
   $record = array(
     'category' => $form_state['values']['category'],
+    'user_status' => serialize($form_state['values']['user_status']),
     'recipients' => $recipients,
     'reply' => $form_state['values']['reply'],
     'weight' => $form_state['values']['weight'],
diff --git a/mass_contact.install b/mass_contact.install
index d1a20bf..1ce8265 100644
--- a/mass_contact.install
+++ b/mass_contact.install
@@ -34,6 +34,11 @@ function mass_contact_schema() {
         'size' => 'big',
         'not null' => TRUE,
       ),
+      'user_status' => array(
+        'description' => '',
+        'type' => 'text',
+        'size' => 'big',
+      ),
       'reply' => array(
         'description' => '.',
         'type' => 'text',
@@ -347,3 +352,16 @@ function mass_contact_update_7101(&$sandbox) {
     db_query('UPDATE {mass_contact} SET recipients = :recipients WHERE cid = :cid', array(':cid' => $record->cid, ':recipients' => $recipients));
   }
 }
+
+/**
+ * Add user_status field.
+ */
+function mass_contact_update_7102() {
+  $spec = array(
+    'type' => 'text',
+    'size' => 'big',
+  );
+  db_add_field('mass_contact', 'user_status', $spec);
+
+  return t('User status field added - mass_contact module.');
+}
diff --git a/mass_contact.page.inc b/mass_contact.page.inc
index 01a67f9..84da8f6 100644
--- a/mass_contact.page.inc
+++ b/mass_contact.page.inc
@@ -48,7 +48,7 @@ function mass_contact_mail_page(array $form, array $form_state) {
     ->orderBy('category', 'ASC')
     ->execute();
   foreach ($result as $category) {
-    if (user_access('mass contact send to users in the ' . $category->category . ' category')) {
+    if (user_access('mass contact send messages')) {
       $categories[$category->cid] = check_plain($category->category);
 
       if ($category->selected) {
@@ -1157,8 +1157,24 @@ function _mass_contact_create_recipient_list($category, $respect_opt_outs) {
 
   global $user;
 
-  $query = "SELECT u.name, u.mail, u.data FROM {users} u WHERE uid IN (:uids) AND u.status <> 0";
-  $results = db_query($query, array(':uids' => $uids));
+  $user_status = unserialize($category->user_status);
+  $status = [];
+  foreach ($user_status as $key => $value) {
+    if ($value !== 0) {
+      if ($key == 'active') {
+        $status[] = 1;
+      } elseif ($key == 'blocked') {
+        $status[] = 0;
+      }
+    }
+  }
+  if (!empty($status)) {
+    $status[] = 1;
+  }
+
+  $query = "SELECT u.name, u.mail, u.data FROM {users} u WHERE uid IN (:uids) AND u.status IN (:status)";
+  $results = db_query($query, array(':uids' => $uids, ':status' => $status));
+
   foreach ($results as $account) {
     $account_data = unserialize($account->data);
 
diff --git a/plugins/mass_contact_role.inc b/plugins/mass_contact_role.inc
index 312f583..7db43f5 100644
--- a/plugins/mass_contact_role.inc
+++ b/plugins/mass_contact_role.inc
@@ -49,12 +49,12 @@ function mass_contact_role_recipients(array $recipients) {
   // Select the users.
   if (!empty($recipients['mass_contact_role'][0]) && $recipients['mass_contact_role'][0] == 2) {
     // If the selected role is the authenticated users role, get everyone.
-    $query = "SELECT u.uid FROM {users} u LEFT JOIN {users_roles} ur ON u.uid = ur.uid WHERE status = 1";
+    $query = "SELECT u.uid FROM {users} u LEFT JOIN {users_roles} ur ON u.uid = ur.uid";
     $result = db_query($query)->fetchAllAssoc('uid', PDO::FETCH_ASSOC);
   }
   elseif (!empty($recipients['mass_contact_role'])) {
     // Get the list of users who belong to the current role.
-    $query = "SELECT u.uid FROM {users} u LEFT JOIN {users_roles} ur ON u.uid = ur.uid WHERE ur.rid IN (:ur_rids) AND status = 1";
+    $query = "SELECT u.uid FROM {users} u LEFT JOIN {users_roles} ur ON u.uid = ur.uid WHERE ur.rid IN (:ur_rids)";
     $result = db_query($query, array(':ur_rids' => $recipients['mass_contact_role']))->fetchAllAssoc('uid', PDO::FETCH_ASSOC);
   }
   if (empty($result)) {
